### PR TITLE
[util] Implement string helpers on non-Windows platforms

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Title.

I think we could probably just use these on Windows too tbh.

### CODE REVIEW

File: src/util/util_string.cpp [hunk 1]
```diff
@@ -15,16 +19,26 @@ namespace dxvk::str {
     ::WideCharToMultiByte(CP_UTF8, 0, ws, -1,
       &result.at(0), len, nullptr, nullptr);
     return result;
+#else
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
```
Comment (doitsujin) [inline src/util/util_string.cpp:23]: This is deprecated in C++17.

**Review by doitsujin — CHANGES_REQUESTED** [review]
See comments.

If we can get rid of the Win32-specific functions altogether I'd prefer that over ifdef sauce, but we need things to work properly and test with somewhat common stuff, such as Japanese game executable names. We could always roll our own UCS-2 or UTF-16 to UTF-8 conversion as well, the encodings are simple enough.

There's also potential issues on platforms where wchar_t is 32-bit.

File: src/util/util_string.cpp (hunk 1 above)
Comment (Joshua-Ashton) [reply src/util/util_string.cpp:23]: I hate the C++ standards committee. *sigh*

Comment (Joshua-Ashton) [reply src/util/util_string.cpp:23]: Wow they really deprecated this and said "no replacement" huh.

Absolutely incredible work.

**Comment by Joshua-Ashton** [comment]
In DXVK Native, it just uses the native wchar size.

**Comment by pchome** [comment]
Any plans to use [str::filename patch](https://github.com/doitsujin/dxvk/commit/1053d5c975ea85cf18b481c977c21963eec20aca) from #2138 ?

**Comment by doitsujin** [comment]
Superseded by #2758.

**Pull request closed without merging**